x-pack/plugin/ent-search/src/main/java/org/elasticsearch/xpack/entsearch/engine/action/ListEnginesAction.java

```diff
@@ -0,0 +1,168 @@
+        }
+
+        public String query() {
+            return query;
+        }
+
+        public PageParams pageParams() {
+            return pageParams;
+        }
+
+        @Override
+        public ActionRequestValidationException validate() {
+            // Validation is done as part of PageParams constructor
+            return null;
+        }
+
+        @Override
+        public void writeTo(StreamOutput out) throws IOException {
+            super.writeTo(out);
+            out.writeOptionalString(query);
+            pageParams.writeTo(out);
+        }
+
+        @Override
+        public String[] indices() {
+            return new String[] { EngineIndexService.ENGINE_ALIAS_NAME };
+        }
+
+        @Override
+        public IndicesOptions indicesOptions() {
+            return IndicesOptions.STRICT_SINGLE_INDEX_NO_EXPAND_FORBID_CLOSED;
+        }
+
+        @Override
+        public boolean equals(Object o) {
+            if (this == o) return true;
+            if (o == null || getClass() != o.getClass()) return false;
+            Request that = (Request) o;
+            return Objects.equals(query, that.query) && Objects.equals(pageParams, that.pageParams);
+        }
+
+        @Override
+        public int hashCode() {
+            return Objects.hash(query, pageParams);
+        }
+    }
+
+    public static class Response extends ActionResponse implements StatusToXContentObject {
+
+        final Engine[] engines;
+        final PageParams pageParams;
+        final Integer totalResults;
+
+        public Response(StreamInput in) throws IOException {
+            super(in);
+            this.pageParams = new PageParams(in);
+            this.totalResults = in.readVInt();
+            this.engines = in.readArray(Engine::new, Engine[]::new);
+        }
+
+        public Response(Engine[] engines, PageParams pageParams, Integer totalResults) {
+            this.engines = engines;
+            this.pageParams = pageParams;
+            this.totalResults = totalResults;
+        }
+
+        @Override
+        public void writeTo(StreamOutput out) throws IOException {
+            pageParams.writeTo(out);
+            out.writeVInt(totalResults);
+            out.writeArray(engines);
+        }
+
+        @Override
+        public XContentBuilder toXContent(XContentBuilder builder, Params params) throws IOException {
+            builder.startObject();
+            {
+                builder.startObject("_meta");
+                {
+                    builder.field("from", pageParams.getFrom());
+                    builder.field("size", pageParams.getSize());
+                    builder.field("total", totalResults);
+                }
+                builder.endObject();
+            }
+
+            builder.array("results", (Object[]) engines);
+            builder.endObject();
+
+            return builder;
+        }
+
+        @Override
+        public RestStatus status() {
```

Comment:
I think we could omit this overridden method?

Reply:
It's part of the `StatusToXContentObject` interface, it needs to be implemented.